libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -78,3 +140,571 @@ void test_MQTT_Init_Invalid_params( void )
+{
+    pNetworkBuffer->pBuffer = mqttBuffer;
+    pNetworkBuffer->size = MQTT_TEST_BUFFER_LENGTH;
+}
+
+/**
+ * @brief Mocked MQTT event callback.
+ */
+static void eventCallback( MQTTContext_t * pContext,
+                           MQTTPacketInfo_t * pPacketInfo,
+                           uint16_t packetIdentifier,
+                           MQTTPublishInfo_t * pPublishInfo )
+{
+    ( void ) pContext;
+    ( void ) pPacketInfo;
+    ( void ) packetIdentifier;
+    ( void ) pPublishInfo;
+}
+
+/**
+ * @brief A mocked timer query function that increments on every call. This
+ * guarantees that only a single iteration runs in the ProcessLoop for ease
+ * of testing.
+ */
+static uint32_t getTime( void )
+{
+    return globalEntryTime++;
+}
+
+/**
+ * @brief Mocked successful transport send.
+ */
+static int32_t transportSendSuccess( MQTTNetworkContext_t pContext,
+                                     const void * pBuffer,
+                                     size_t bytesToWrite )
+{
+    ( void ) pContext;
+    ( void ) pBuffer;
+    return bytesToWrite;
+}
+
+/**
+ * @brief Mocked successful transport read.
+ */
+static int32_t transportRecvSuccess( MQTTNetworkContext_t pContext,
+                                     void * pBuffer,
+                                     size_t bytesToRead )
+{
+    ( void ) pContext;
+    ( void ) pBuffer;
+    return bytesToRead;
+}
+
+/**
+ * @brief Initialize the transport interface with the mocked functions for
+ * send and receive.
+ */
+static void setupTransportInterface( MQTTTransportInterface_t * pTransport )
+{
+    pTransport->networkContext = 0;
+    pTransport->send = transportSendSuccess;
+    pTransport->recv = transportRecvSuccess;
+}
+
+/**
+ * @brief Initialize our event and time callback with the mocked functions
+ * defined for the purposes this test.
+ */
+static void setupCallbacks( MQTTApplicationCallbacks_t * pCallbacks )
+{
+    pCallbacks->appCallback = eventCallback;
+    pCallbacks->getTime = getTime;
+}
+
+/**
+ * @brief MQTT_GetIncomingPacketTypeAndLength callback used by CMock for setting
+ * the type of the incoming packet to the same value as currentPacketType then
+ * returns modifyIncomingPacketStatus (global variables).
+ */
+static MQTTStatus_t modifyIncomingPacket( MQTTTransportRecvFunc_t readFunc,
+                                          MQTTNetworkContext_t networkContext,
+                                          MQTTPacketInfo_t * pIncomingPacket,
+                                          int cmock_num_calls )
+{
+    /* Remove unused parameter warnings. */
+    ( void ) readFunc;
+    ( void ) networkContext;
+    ( void ) cmock_num_calls;
+
+    pIncomingPacket->type = currentPacketType;
+    return modifyIncomingPacketStatus;
+}
+
+/**
+ * @brief MQTT_GetPingreqPacketSize callback used by CMock for setting the size
+ * of a PINGREQ packet and returns successfully.
+ */
+static MQTTStatus_t modifyPacketSize( size_t * pPacketSize,
```

Comment:
This function doesn't seem necessary to me. You can just use

```
size_t pingreqSize = 2;
MQTT_GetPingreqPacketSize_ExpectAnyArgsAndReturn( MQTTSuccess );
MQTT_GetPingreqPacketSize_ReturnThruPtr_pPacketSize( &pingreqSize );
```

at line 296 where the call to this function is made.